uint16 disl1
uint16 disl2
uint16 disl3
uint16 disl4
uint16 disl5